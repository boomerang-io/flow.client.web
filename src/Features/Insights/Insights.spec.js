/* eslint-disable jest/no-commented-out-tests */
import React from "react";
import WorkflowInsights from "./index";
import { startApiServer } from "ApiServer";
import { queryCaches } from "react-query";

jest.mock("@carbon/charts-react", () => ({
  ...jest.requireActual("@carbon/charts-react"),
  DonutChart: "DonutChart",
  LineChart: "LineChart",
  ScatterChart: "ScatterChart",
}));

// const props = {
//   teams: {
//     isFetching: false,
//     status: "success",
//     error: "",
//     data: [],
//   },
//   insights: {
//     isFetching: false,
//     status: "success",
//     error: "",
//     data: {
//       totalActivitiesExecuted: 107,
//       medianExecutionTime: 24435,
//       executions: [
//         {
//           creationDate: 1535760000000,
//           duration: 20825,
//           id: "5bed8335b336a5000103aa2b",
//           initiatedByUserId: "5bbcdd38ee763e00011f5b72",
//           initiatedByUserName: "Adrienne Hudson",
//           status: "completed",
//           workflowId: "5bed82f6b336a5000103aa1d",
//           workflowRevisionid: "5bed82f7b336a5000103aa20",
//           description: "test delete",
//           icon: "flow",
//           shortDescription: "test",
//           workflowName: "Adrienne3",
//           teamName: "CAI Offering Team",
//         },
//       ],
//     },
//   },
// };

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
  queryCaches.forEach((queryCache) => queryCache.clear());
});

describe("WorkflowInsights --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowInsights", async () => {
    const { baseElement, findByTestId } = rtlContextRouterRender(<WorkflowInsights />);
    await findByTestId("completed-insights");
    const a11yElement = baseElement.querySelector("#a11y-status-message");
    if (baseElement.contains(a11yElement)) {
      a11yElement.parentNode.removeChild(a11yElement);
    }
    expect(baseElement).toMatchSnapshot();
  });
});

//New tests failed due to some feature on @carbon/carts - Check again when this lib get stable
// const insights = {
//   totalActivitiesExecuted: 107,
//   medianExecutionTime: 24435,
//   executions: [
//     {
//       creationDate: 1542292277199,
//       duration: 20825,
//       id: "5bed8335b336a5000103aa2b",
//       initiatedByUserId: "5bbcdd38ee763e00011f5b72",
//       initiatedByUserName: "Adrienne Hudson",
//       status: "completed",
//       workflowId: "5bed82f6b336a5000103aa1d",
//       workflowRevisionid: "5bed82f7b336a5000103aa20",
//       description: "test delete",
//       icon: "flow",
//       shortDescription: "test",
//       workflowName: "Adrienne3",
//       teamName: "CAI Offering Team"
//     }
//   ]
// }
// const initialReduxState ={
//   insights: initialState,
//   teams: {
//     isFetching: false,
//     status: "success",
//     error: "",
//     data: [
//       {
//         "id": "5678",
//         "name": "Isa's Team",
//         "workflows": [
//           {
//             "id": "111",
//             "name": "isabela-workflow-1",
//             "shortDescription": "blablabla",
//             "status": "published",
//             "icon": "mail",
//             "flowTeamId": "5678",
//             "inputs": [
//               {
//                 "description": "A boolean parameter",
//                 "key": "toggle.bool",
//                 "label": "Boolean Toggle",
//                 "type": "boolean",
//                 "value": true
//               },
//               {
//                 "description": "A select parameter",
//                 "key": "select.options",
//                 "label": "Select Options",
//                 "type": "select",
//                 "isRequired": true,
//                 "value": ["op1", "op2"]
//               }
//             ],
//             "properties": [
//               {
//                 "readOnly": false,
//                 "description": "Please help me",
//                 "key": "what.this",
//                 "label": "Boomerang incoming webhook",
//                 "type": "boolean",
//                 "defaultValue": "false",
//                 "required": true
//               }
//             ]
//           },
//           {
//             "id": "222",
//             "name": "isabela-workflow-2",
//             "shortDescription": "blablabla",
//             "status": "draft",
//             "icon": "schedule",
//             "flowTeamId": "5678",
//             "inputs": [
//               {
//                 "description": "A boolean parameter",
//                 "key": "toggle.bool",
//                 "label": "Boolean Toggle",
//                 "type": "boolean",
//                 "value": false
//               },
//               {
//                 "description": "A select parameter",
//                 "key": "select.options",
//                 "label": "Select Options",
//                 "type": "select",
//                 "isRequired": true,
//                 "value": []
//               }
//             ]
//           }
//         ]
//       }
//     ]
//   }
// }
// const props = {
//   location: {},
//   insightsActions,
//   // teamsActions: {
//   //   fetch: () => new Promise(() => {})
//   // },
// };
// beforeEach(() => {
//   mockAxios.resetHistory();
//   window.SVGElement.prototype.getComputedTextLength = () => 200;
// });
// describe("WorkflowInsights --- Snapshot", () => {
//   it("Capturing Snapshot of WorkflowInsights", async () => {
//     const url = new RegExp(`${BASE_URL}/insights?`);
//     mockAxios.onGet(url).reply(200, insights);
//     const { baseElement, findByText } = rtlContextRouterRender(<WorkflowInsights {...props} />,{
//       initialState:initialReduxState
//     });
//     await findByText(/Filter by Workflow/);
//     expect(baseElement).toMatchSnapshot();
//   });
// });

// describe("WorkflowInsights --- RTL", () => {
//   it("Test Team and Period Filters", async () => {
//     const url = new RegExp(`${BASE_URL}/insights?`);
//     mockAxios.onGet(url).reply(200, insights);
//     const { findByText, getByText, getByLabelText, getAllByText, getByTestId} = rtlContextRouterRender(<WorkflowInsights {...props} />,{
//       initialState:initialReduxState
//     });
//     await findByText(/Filter by Workflow/);
//     const teamFilter = getByLabelText("Filter by Team");
//     fireEvent.click(teamFilter);
//     fireEvent.click(teamFilter.firstChild);
//     await wait(() => {
//       expect(mockAxios.history.get.length).toBe(2);
//     });
//     const periodFilter = getByLabelText("Time period");
//     fireEvent.click(periodFilter);
//     fireEvent.click(getByText(/1 day/));
//     await wait(() => {
//       expect(mockAxios.history.get.length).toBe(3);
//     });
//   });
// });
