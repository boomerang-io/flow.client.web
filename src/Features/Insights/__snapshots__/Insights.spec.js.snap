// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowInsights --- Snapshot Capturing Snapshot of WorkflowInsights 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__header"
        >
          <h1
            class="bx--bmrg-feature-header-title"
          >
            Insights
          </h1>
          <p
            class="bx--bmrg-feature-header-title"
          >
            View your Worfklow execution stats
          </p>
        </div>
      </section>
    </header>
    <div
      class="container"
    >
      <div
        class="dataFilters"
      >
        <div
          class="bx--multi-select__wrapper bx--list-box__wrapper"
        >
          <label
            class="bx--label"
            for="actions-scopes-select-input"
            id="actions-scopes-select-label"
          >
            Filter by scope
          </label>
          <div
            class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
          >
            <div
              class="bx--list-box__field"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="bx--text-input bx--text-input--empty"
                id="actions-scopes-select-input"
                placeholder="Choose scope(s)"
                role="combobox"
                value=""
              />
              
              <button
                aria-haspopup="true"
                aria-label="Open"
                class="bx--list-box__menu-icon"
                data-toggle="true"
                role="button"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bx--multi-select__wrapper bx--list-box__wrapper"
        >
          <label
            class="bx--label"
            for="insights-teams-select-input"
            id="insights-teams-select-label"
          >
            Filter by Team
          </label>
          <div
            class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
          >
            <div
              class="bx--list-box__field"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="bx--text-input bx--text-input--empty"
                id="insights-teams-select-input"
                placeholder="Choose team(s)"
                role="combobox"
                value=""
              />
              
              <button
                aria-haspopup="true"
                aria-label="Open"
                class="bx--list-box__menu-icon"
                data-toggle="true"
                role="button"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bx--multi-select__wrapper bx--list-box__wrapper"
        >
          <label
            class="bx--label"
            for="insights-workflows-select-input"
            id="insights-workflows-select-label"
          >
            Filter by Workflow
          </label>
          <div
            class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
          >
            <div
              class="bx--list-box__field"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="bx--text-input bx--text-input--empty"
                id="insights-workflows-select-input"
                placeholder="Choose workflow(s)"
                role="combobox"
                value=""
              />
              
              <button
                aria-haspopup="true"
                aria-label="Open"
                class="bx--list-box__menu-icon"
                data-toggle="true"
                role="button"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bx--multi-select__wrapper bx--list-box__wrapper"
        >
          <label
            class="bx--label"
            for="insights-triggers-select-input"
            id="insights-triggers-select-label"
          >
            Filter by trigger
          </label>
          <div
            class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
          >
            <div
              class="bx--list-box__field"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="bx--text-input bx--text-input--empty"
                id="insights-triggers-select-input"
                placeholder="Choose trigger type(s)"
                role="combobox"
                value=""
              />
              
              <button
                aria-haspopup="true"
                aria-label="Open"
                class="bx--list-box__menu-icon"
                data-toggle="true"
                role="button"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="bx--multi-select__wrapper bx--list-box__wrapper"
        >
          <label
            class="bx--label"
            for="insights-statuses-select-input"
            id="insights-statuses-select-label"
          >
            Filter by status
          </label>
          <div
            class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
          >
            <div
              class="bx--list-box__field"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="bx--text-input bx--text-input--empty"
                id="insights-statuses-select-input"
                placeholder="Choose status(es)"
                role="combobox"
                value=""
              />
              
              <button
                aria-haspopup="true"
                aria-label="Open"
                class="bx--list-box__menu-icon"
                data-toggle="true"
                role="button"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="timeFilters"
        >
          <div
            class="bx--form-item"
          >
            <div
              class="bx--date-picker bx--date-picker--range"
              id="insights-date-picker"
            >
              <div
                class="bx--date-picker-container"
              >
                <label
                  class="bx--label"
                  for="insights-date-picker-start"
                >
                  Start date
                </label>
                <div
                  class="bx--date-picker-input__wrapper"
                >
                  <input
                    autocomplete="off"
                    class="bx--date-picker__input flatpickr-input"
                    id="insights-date-picker-start"
                    pattern="\\\\d{1,2}\\\\/\\\\d{1,2}\\\\/\\\\d{4}"
                    type="text"
                    value="01/01/2020"
                  />
                  <svg
                    aria-hidden="true"
                    class="bx--date-picker__icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="bx--date-picker-container"
              >
                <label
                  class="bx--label"
                  for="insights-date-picker-end"
                >
                  End date
                </label>
                <div
                  class="bx--date-picker-input__wrapper"
                >
                  <input
                    autocomplete="off"
                    class="bx--date-picker__input"
                    data-fp-omit=""
                    id="insights-date-picker-end"
                    pattern="\\\\d{1,2}\\\\/\\\\d{1,2}\\\\/\\\\d{4}"
                    type="text"
                    value="01/01/2020"
                  />
                  <svg
                    aria-hidden="true"
                    class="bx--date-picker__icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="statsWidgets"
        data-testid="completed-insights"
      >
        <div
          class="bx--tile container"
        >
          <div>
            <h3
              class="title"
            >
              Executions
            </h3>
            <div
              class="totalCountContainer"
            >
              <p
                class="totalCount"
              >
                1088
              </p>
              <p
                class="type"
              >
                runs
              </p>
            </div>
          </div>
          <ul
            class="list"
          >
            <li
              class="row"
            >
              <p
                class="itemValue"
                style="width: 2rem;"
              >
                4
              </p>
              <p
                class="itemLabel"
              >
                Tim Testing Slacks
              </p>
            </li>
          </ul>
        </div>
        <div
          class="bx--tile container"
        >
          <div>
            <h3
              class="title"
            >
              Duration (median)
            </h3>
            <div
              class="totalCountContainer"
            >
              <p
                class="totalCount"
              >
                1 min 7 secs
              </p>
              <p
                class="type"
              />
            </div>
          </div>
          <ul
            class="list"
          >
            <li
              class="row"
            >
              <p
                class="itemValue"
                style="width: 7rem;"
              >
                10 secs
              </p>
              <p
                class="itemLabel"
              >
                Minimum
              </p>
            </li>
            <li
              class="row"
            >
              <p
                class="itemValue"
                style="width: 7rem;"
              >
                30 secs
              </p>
              <p
                class="itemLabel"
              >
                Maximum
              </p>
            </li>
            <li
              class="row"
            >
              <p
                class="itemValue"
                style="width: 7rem;"
              >
                23 secs
              </p>
              <p
                class="itemLabel"
              >
                Average
              </p>
            </li>
          </ul>
        </div>
        <div
          class="donut"
        >
          <div
            class="CarbonDonutChart"
          >
            <div>
              DonutChart
            </div>
          </div>
        </div>
      </div>
      <div
        class="graphsWidgets"
      >
        <div
          class="bx--tile container"
        >
          <div>
            LineChart
          </div>
        </div>
        <div
          class="bx--tile container"
        >
          <div>
            ScatterChart
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-label="calendar-container"
    class="flatpickr-calendar rangeMode animate bx--date-picker__calendar"
    role="region"
    tabindex="-1"
  >
    <div
      class="flatpickr-months"
    >
      <span
        class="flatpickr-prev-month"
      >
        
      
        <svg
          height="16px"
          viewBox="0 0 16 16"
          width="16px"
        >
          
        
          <polygon
            points="5,8 10,3 10.7,3.7 6.4,8 10.7,12.3 10,13 "
          />
          
        
          <rect
            height="16"
            style="fill:none"
            width="16"
          />
          
      
        </svg>
      </span>
      <div
        class="flatpickr-month bx--date-picker__month"
      >
        <div
          class="flatpickr-current-month"
        >
          <span
            class="cur-month"
          >
            January
          </span>
          <div
            class="numInputWrapper"
          >
            <input
              aria-label="Year"
              class="numInput cur-year"
              max="2020"
              tabindex="-1"
              type="number"
            />
            <span
              class="arrowUp"
            />
            <span
              class="arrowDown"
            />
          </div>
        </div>
      </div>
      <span
        class="flatpickr-next-month flatpickr-disabled"
      >
        
      
        <svg
          height="16px"
          viewBox="0 0 16 16"
          width="16px"
        >
          
        
          <polygon
            points="11,8 6,13 5.3,12.3 9.6,8 5.3,3.7 6,3 "
          />
          
        
          <rect
            height="16"
            style="fill:none"
            width="16"
          />
          
      
        </svg>
      </span>
    </div>
    <div
      class="flatpickr-innerContainer"
    >
      <div
        class="flatpickr-rContainer"
      >
        <div
          class="flatpickr-weekdays bx--date-picker__weekdays"
        >
          <div
            class="flatpickr-weekdaycontainer"
          >
            
      
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              S
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              M
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              T
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              W
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              Th
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              F
            </span>
            <span
              class="flatpickr-weekday bx--date-picker__weekday"
            >
              S
            </span>
            
      
          </div>
        </div>
        <div
          class="flatpickr-days bx--date-picker__days"
          tabindex="-1"
        >
          <div
            class="dayContainer"
          >
            <span
              aria-label="December 29, 2019"
              class="flatpickr-day prevMonthDay bx--date-picker__day"
              tabindex="-1"
            >
              29
            </span>
            <span
              aria-label="December 30, 2019"
              class="flatpickr-day prevMonthDay bx--date-picker__day"
              tabindex="-1"
            >
              30
            </span>
            <span
              aria-label="December 31, 2019"
              class="flatpickr-day prevMonthDay bx--date-picker__day"
              tabindex="-1"
            >
              31
            </span>
            <span
              aria-label="January 1, 2020"
              class="flatpickr-day selected startRange endRange bx--date-picker__day"
              tabindex="-1"
            >
              1
            </span>
            <span
              aria-label="January 2, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              2
            </span>
            <span
              aria-label="January 3, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              3
            </span>
            <span
              aria-label="January 4, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              4
            </span>
            <span
              aria-label="January 5, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              5
            </span>
            <span
              aria-label="January 6, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              6
            </span>
            <span
              aria-label="January 7, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              7
            </span>
            <span
              aria-label="January 8, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              8
            </span>
            <span
              aria-label="January 9, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              9
            </span>
            <span
              aria-label="January 10, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              10
            </span>
            <span
              aria-label="January 11, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              11
            </span>
            <span
              aria-label="January 12, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              12
            </span>
            <span
              aria-label="January 13, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              13
            </span>
            <span
              aria-label="January 14, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              14
            </span>
            <span
              aria-label="January 15, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              15
            </span>
            <span
              aria-label="January 16, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              16
            </span>
            <span
              aria-label="January 17, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              17
            </span>
            <span
              aria-label="January 18, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              18
            </span>
            <span
              aria-label="January 19, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              19
            </span>
            <span
              aria-label="January 20, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              20
            </span>
            <span
              aria-label="January 21, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              21
            </span>
            <span
              aria-label="January 22, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              22
            </span>
            <span
              aria-label="January 23, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              23
            </span>
            <span
              aria-label="January 24, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              24
            </span>
            <span
              aria-label="January 25, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              25
            </span>
            <span
              aria-label="January 26, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              26
            </span>
            <span
              aria-label="January 27, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              27
            </span>
            <span
              aria-label="January 28, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              28
            </span>
            <span
              aria-label="January 29, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              29
            </span>
            <span
              aria-label="January 30, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              30
            </span>
            <span
              aria-label="January 31, 2020"
              class="flatpickr-day flatpickr-disabled bx--date-picker__day"
            >
              31
            </span>
            <span
              aria-label="February 1, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              1
            </span>
            <span
              aria-label="February 2, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              2
            </span>
            <span
              aria-label="February 3, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              3
            </span>
            <span
              aria-label="February 4, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              4
            </span>
            <span
              aria-label="February 5, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              5
            </span>
            <span
              aria-label="February 6, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              6
            </span>
            <span
              aria-label="February 7, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              7
            </span>
            <span
              aria-label="February 8, 2020"
              class="flatpickr-day nextMonthDay flatpickr-disabled bx--date-picker__day"
            >
              8
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
